### PR TITLE
feat: add kubernetes context in the query

### PR DESCRIPTION
**Pull request by simonrdz**
# Description

Added kubernetes context passing to be able to avoid context targeting error when using this data module when getting secrets. So now context like dev.test or prod.test have to be passed via the module variables to get the data then fill the RDS module.

```
module "rds-postgres-password" {
  source    = "github.com/jobteaser/terraform-kubernetes-get-secret?ref=v0.2.3"

  namespace = "default"
  name = "terraform"
  key = "database-password"
  context = "<env>.test"
}
```

# References

# Motivation and Context

Actually this module is used to get specific secrets from kubernetes then provision a RDS instance when used in a stack. This use the kube context you're on within your session so if your context is dev.test and you apply a stack for prod.test you will create a db or update the prod with the dev credential it is a big source of mistakes with possibility hudges consequencies. 

# How has this been tested

Create a json file to simulate a call from a module

```
{
    "namespace": "test",
    "key": "password",
    "context": "dev.test",
    "name": "test-db-admin"
}
```
ctx to prod.test on my terminal session ```kubectx prod.test```

Then echo the file to the main.go like the following to get secret from the dev

```
cat test.json | go run main.go
Connecting to Kubernetes Context dev.test
{
  "value": "xxxxxxxxxxxxxx"
}
```

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist

- [ ] This PR requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] My code is formatted using terraform fmt.
- [x] There is a **README.md** file in my module (I've created/updated if needed) with a sample usage for this module

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,14 +63,16 @@ Available outputs are listed below, along with their description:
...
-$ git clone git@github.com:gearnode/terraform-kubernetes-get-secret.git
+$ git clone git@github.com:jobteaser/terraform-kubernetes-get-secret.git
 $ cd terraform-kubernetes-get-secret
 $ make
 ```
 
 ## Maintainers
 
-This module is currently maintained by the individuals listed below.
+This module is currently a fork maintained by the individuals listed below.
```
Comment (jt-enrico) [inline README.md]:
```suggestion
This module is a fork of [gearnode/terraform-kubernetes-get-secret](https://github.com/gearnode/terraform-kubernetes-get-secret).
```

Comment (jt-enrico) [reply README.md]: And remove the list below :)

**Review by jt-enrico — COMMENTED** [review]
I am not sure we can/should remove the binaries.

**Comment by simonrdz** [comment]
> I am not sure we can/should remove the binaries.

Updated but not removed

File: README.md [hunk 2]
```diff
@@ -61,14 +63,16 @@ Available outputs are listed below, along with their description:
...
 
 - [Bryan Frimin](https://github.com/gearnode)
 - [Ludovic Vielle](https://github.com/lukkor)
```
Comment (simonrdz) [inline README.md]:
```suggestion
```

**Review by gearnode — COMMENTED** [review]
- use go mod to manage dependencies. 
- you must have binary to call it from terraform

**Comment by simonrdz** [comment]
> * use go mod to manage dependencies.
> * you must have binary to call it from terraform

I used the Makefile to build the binaries then pushed them do I need to do other stuff ?

**Comment by gearnode** [comment]
Your feature request is available on the master https://github.com/gearnode/terraform-kubernetes-get-secret

**Pull request closed without merging**